crates/noirc_frontend/src/hir/resolution/resolver.rs

```diff
@@ -352,6 +340,26 @@ impl<'a> Resolver<'a> {
         }
     }
 
+    fn resolve_fixed_variable_array_length(&mut self, name: Ident) -> Type {
+        // A resolved identifier must exist either in the local scope or global scope
+        let hir_ident = self.find_variable(&name);
+        let definition_info = self.interner.definition(hir_ident.id);
+        if definition_info.mutable {
+            self.push_err(ResolverError::ExpectedConstVariable {
+                name: name.0.contents.clone(),
+                span: name.span(),
+            });
+            return Type::Error;
+        }
+
+        if let Some(rhs_expr_id) = definition_info.rhs {
+            let length = self.get_fixed_variable_array_length(&rhs_expr_id);
+            Type::ArrayLength(length)
+        } else {
+            Type::Error
```

Comment:
We should issue an error to the user in this case, otherwise this error type does nothing but prevent/cause future errors.

Reply:
I made an error for the user, and let me know what you think. However, I am realizing that if we are able to find a variable but it does not have a rhs expr stored that is a greater issue on Noir's end rather than the users. I don't want to just panic, but I was unsure entirely what to label the error as it seems like something that is a failure in our resolution logic rather than something the user did.

Reply:
A variable not having an rhs is a valid case and occurs most often with parameters - though parameters are not valid in a type's array length position anyway